app/scripts/modules/appengine/src/serverGroup/configure/wizard/ConfigFileArtifactList.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+
+import { IArtifact, IExpectedArtifact, IPipeline, IStage, StageArtifactSelectorDelegate } from '@spinnaker/core';
+
+interface IConfigFileArtifactListProps {
+  configArtifacts: any[];
+  pipeline: IPipeline;
+  stage: IStage;
+  updateConfigArtifacts: (configArtifacts: any[]) => void;
+  updatePipeline: (changes: Partial<IPipeline>) => void;
+}
+
+export const ConfigFileArtifactList = (props: IConfigFileArtifactListProps) => {
+  const addConfigArtifact = () => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.push({});
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  const deleteConfigArtifact = (index: number) => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.splice(index, 1);
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  const onExpectedArtifactEdited = (artifact: IArtifact, index: number): void => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.splice(index, 1, { ...newConfigArtifacts[index], id: null, artifact });
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  const onExpectedArtifactSelected = (expectedArtifact: IExpectedArtifact, index: number): void => {
+    onChangeExpectedArtifactId(expectedArtifact.id, index);
+  };
+
+  const onChangeExpectedArtifactId = (id: string, index: number): void => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.splice(index, 1, { ...newConfigArtifacts[index], id, artifact: null });
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  const onExpectedArtifactAccountSelected = (account: string, index: number): void => {
+    const newConfigArtifacts = [...props.configArtifacts];
+    newConfigArtifacts.splice(index, 1, { ...newConfigArtifacts[index], account });
+    props.updateConfigArtifacts(newConfigArtifacts);
+  };
+
+  return (
+    <>
+      {props.configArtifacts.map((a, i) => {
+        return (
+          <div
+            key={i}
+            className={
+              'artifact-configuration-section col-md-12' + (props.configArtifacts.length - 1 === i ? 'last-entry' : '')
```

Comment:
maybe cleaner to use [classnames](https://github.com/JedWatson/classnames) here?

Reply:
👍